text/0000-argument-lifetimes.md

```diff
@@ -0,0 +1,492 @@
+- Feature Name: argument_lifetimes
+- Start Date: 2017-08-17
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Improves the clarity, ergonomics, and learnability around explicit lifetimes, so
+that instead of writing
+
+```rust
+fn two_args<'a, 'b>(arg1: &'a Foo, arg2: &'b Bar) -> &'b Baz
```

Comment:
Why did you write `<'a, 'b>` here when the `'a` has no reason to be named, but leave the lifetime unnamed on `&Foo` below?

The way you've currently written it makes the impact of the binding site more significant than it needs to be (which might be great for motivating the RFC, but lets not do it on false pretenses...).

It would be more fair IMO to make the above signature: `fn two_args<'b>(arg1: &Foo, arg2: &'b Bar) -> &'b Baz`.